infra/modules/network/main.tf

```diff
@@ -14,13 +14,16 @@ module "aws_vpc" {
   azs  = local.availability_zones
   cidr = local.vpc_cidr
 
+  # Database subnet is only created if a database is needed
+  create_database_subnet_group = var.has_database
```

Comment:
nit: move this closer to the `database_subnet_group_name` line since it is only associated with that line. this flag doesn't actually refer to creation of the subnets themselves. a "subnet group" is a specific thing that means more than a group/set of subsets (AWS is confusing).